mobile/src/main/java/com/gelakinetic/mtgfam/helpers/database/CardDbAdapter.java

```diff
@@ -1249,12 +1249,15 @@ public static Cursor Search(SearchCriteria criteria, boolean backface, String[]
                     + " " + criteria.cmc + ")";
         }
 
-        if (criteria.noTokens) {
+        if (criteria.moJhoStoFilter) {
+            /* Filter out tokens. */
             statement += " AND (" +
-                        "(NOT " + DATABASE_TABLE_CARDS + "." + KEY_MANACOST + " = '') " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_SUPERTYPE + " LIKE '%Land%' " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_ABILITY + " LIKE '%Suspend%' " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_ABILITY + " LIKE '%Splice onto Arcane%')";
+                    /* Cards without mana costs. */
+                    "NOT " + DATABASE_TABLE_CARDS + "." + KEY_MANACOST + " = '' " +
+                    /* Cards like 'Dryad Arbor'. */
+                    "OR " + DATABASE_TABLE_CARDS + "." + KEY_SUPERTYPE + " LIKE '%Land Creature%')";
+            /* Filter out 'UN-'sets*/
+            statement += " AND NOT " + DATABASE_TABLE_CARDS + "." + KEY_SET + " IN ('UG', 'UNH')";
```

Comment:
For some reason the B.F.M. was not returned as a random creature with CMC=15, so I assumed incorrectly that 'UN'-sets were filtered out already.